Source/DafnyCore/AST/Tokens.cs

```diff
@@ -244,16 +250,13 @@ public class BoogieRangeToken : TokenWrapper {
   }
 }
 
-public class CodeActionRange : TokenWrapper {
-  public int column;
-  public int length;
-  public CodeActionRange(IToken sourceTok, int column, int length) : base(sourceTok) {
-    this.column = column;
-    this.length = length;
-  }
+public class TokenRange : TokenWrapper {
```

Comment:
> Can you please justify why we have "RangeToken" and "TokenRange" which both contain a start and end token? I know that "TokenRange" is a TokenWrapper whereas "RangeToken" is not, but I feel that they are too close to each other so that having both has a benefit.

Removed one.

> Moreover, previously, it was possible to distinguish code actions that included or excluded the last token, which seems not encoded anymore.

I've changed the semantics of RangeToken so that if EndToken is null, it's only the starting token that defines the range, which seems sufficient.

Reply:
Can you please reassure me?
Until before this PR, a RangeToken was always inclusive from the start of the StartToken to the end of the EndToken. This is what makes it possible to highlight in the IDE an entire expression.

From what you say, it looks like the new semantics are that, if the RangeToken does not have an end, then the startToken defines the beginning and the end inclusive, but if it has an end, then the end of the range is the start of the EndToken
Am I missing something? If this is true, how do you make distinction between the two?

Reply:
If there's an StartToken and EndToken, then the semantics are unchanged: the range will include the whole EndToken.